projects/fsastorefrontlib/src/cms-components/form/cms-form-submit/cms-form-submit.component.ts

```diff
@@ -17,34 +14,9 @@ import { CmsFormSubmitComponent } from '../../../occ/occ-models';
 export class CMSFormSubmitComponent extends FormCMSComponent implements OnInit {
   constructor(
     protected componentData: CmsComponentData<CmsFormSubmitComponent>,
-    protected activatedRoute: ActivatedRoute,
-    protected cmsComponentConnector: CmsComponentConnector,
     protected formDataService: FormDataService,
     protected formDataStorageService: FormDataStorageService
   ) {
     super(componentData, formDataService, formDataStorageService);
   }
-
-  routeParamId = 'formCode';
-  pageContext: PageContext;
-
-  ngOnInit() {
-    this.subscription.add(
-      this.activatedRoute.params
-        .pipe(
-          map(routeParam => {
-            this.pageContext = new PageContext(
-              routeParam[this.routeParamId],
-              PageType.CATEGORY_PAGE
-            );
-            return (this.component$ = this.cmsComponentConnector.get(
-              this.componentData.uid,
-              this.pageContext
-            ));
-          })
-        )
-        .subscribe()
-    );
-    this.loadForm();
```

Comment:
Just to clarify this one. Almost the entire onInit logic is already defined in FormCMSComponent which is extended by CMSFormSubmitComponent. Also, we don't need to fetch component as formDefinition (which is set in parent class) already carries all data needed to render the form.
If anyone remembers why was category and page context needed here, please comment :)